Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift

```diff
@@ -227,6 +273,26 @@ open class ChatChannelListViewModel: ObservableObject, ChatChannelListController
             .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
             .filter { $0.id != chatClient.currentUserId }
     }
+    
+    private func updateSearchResults() {
+        guard let messageSearchController = messageSearchController else {
+            return
+        }
+
+        searchResults = messageSearchController.messages
+            .compactMap { message in
+                if let channelId = message.cid,
+                   let channel = self.chatClient.channelController(for: channelId).channel {
+                    let searchResult = ChannelSelectionInfo(
+                        channel: channel,
+                        message: message
+                    )
+                    return searchResult
+                } else {
+                    return nil
+                }
```

Comment:
There's lot of logic inside of the `compactMap`. Maybe that is worth refactoring a little bit do make it a little more functional.

Reply:
Updated.